lib/applications/send.js

```diff
@@ -0,0 +1,189 @@
+ * @property {number} response - statusCode response from request
+ */
+
+/*
+ * @typedef options
+ * @type object
+ * Application options is the object passed between each of these functions
+ *
+ * @property {string} trigger - one of live/sunset/updated
+ * @property {array} apps - applications pulled from database
+ * @property {array} endpoints - endpoints derived from apps (first added by `endpoints` function)
+ * @property {string} endpoints.id - id of a given app (added by `endpoints` function)
+ * @property {requestEndpoint} endpoints.options - see requestEndpoint typedef (added by `endpoints` function)
+ * @property {endpointResponse} endpoints.response - see endpointResponse typedef (added during `send` Promise.resolve)
+ */
+
+/**
+ * Creates an array from all application endpoints that match the trigger
+ *
+ * @param  {options} options - see options typedef
+ *
+ * @returns {options}  - see options typedef
+ */
+const endpoints = options => {
+  const opts = options;
+
+  return new Promise((resolve, reject) => {
+    if (!Array.isArray(options.apps)) {
+      reject('Apps must be an array');
+    }
+
+    opts.endpoints = options.apps.map(app => {
+      if (app[`${options.trigger}-endpoint`] && app[`${options.trigger}-endpoint`] !== '') {
+        return {
+          id: app.id,
+          options: {
+            url: app[`${options.trigger}-endpoint`],
+            method: 'POST',
+            json: {
+              id: app['client-id'],
+              secret: app['client-secret'],
+            },
+          },
+        };
+      }
+
+      return null;
+    }).filter(Boolean);
+
+    resolve(opts);
+  });
+};
+
+/*
+ * Promise wrapper for Request
+ */
+const req = options => {
+  return new Promise((res, rej) => {
+    request(options, (err, response) => {
+      let status;
+      if (err) {
+        rej(err);
+      }
+
+      if (typeof response === 'object' && response.hasOwnProperty('statusCode')) {
+        status = response.statusCode;
+      }
+
+      res({
+        response: status || 500,
+        timestamp: moment().unix(),
+      });
+    });
+  });
+};
+
+/*
+ * Send request to endpoints
+ *
+ * @param {requestEndpoint} options - see requestEndpoint typedef
+ *
+ * @returns {endpointResponse} - see endpointResponse typedef
+ */
+const requests = options => {
+  const opts = options;
+
+  // grabs every promise for every endpoint
+  const ends = opts.endpoints.map(en => {
+    const end = en;
+
+    return req(end.options).then(resp => {
+      end.response = resp;
+
+      return end;
+    });
+  });
+
+  return Promise.all(ends).then(result => {
+    opts.endpoints = result;
+
+    return opts;
+  });
+};
+
+/**
+ * Updates database with responses from each endpoint
+ *
+ * @param  {options} options - see options typedef
+ *
+ * @returns {promise}  - knex transactions
+ */
+const save = options => {
+  const ids = options.endpoints.map(end => {
+    return end.id;
+  });
+
+  return database.transaction(trx => {
```

Comment:
Why are we using `transaction` here? This is the first I've seen of it in our codebase